js_src/app.js

```diff
@@ -166,8 +178,10 @@ app.changeSoftwareVersion = function(element) {
       goog.dom.appendChild(tr, app.newTD('<a href="' + oflLink + '">View in Open Fixture Library</a>'));
       goog.dom.appendChild(tbody, tr);
     }
-    app.showBlock(personality_fieldset);
-  } else {
+    if (personality_fieldset) {
```

Comment:
I suspect these may be clearer if they're done as "if (personality_fieldset) { if (test) show or hide }", or given it's probably pointless populating the tbody if the fieldset doesn't exist, just test for that as an overall thing first. Equally should we be checking if tbody exists too, or is that not a warning?

Reply:
`tbody` didn't raise a warning. It actually was because I added the explicit not-null `{!Element}` JSDoc type in the show / hide functions and the `goog.dom.$` returns a nullable `Element`. I agree on the clearer style though.

Reply:
Or shall we do the check in the show / hide functions themselves?

Reply:
Doing it in the show/hide sounds sensible, although my gut feeling is that adding " if (personality_fieldset) {" as an initial statement outside the existing if may be clearer still, so we don't process tbody first.

Reply:
Ok, it's fixed now.